components/veml7700/veml7700.h

```diff
@@ -0,0 +1,155 @@
+#define VEML7700_PERSISTENCE_PROTECTION_4 (0b10)
+#define VEML7700_PERSISTENCE_PROTECTION_8 (0b11)
+
+/**
+ * VEML configuration descriptor
+ */
+typedef struct
+{
+    uint16_t gain : 2;                //!< control the sensitivity
+    uint16_t integration_time : 4;    //!< time to measure
+    uint16_t persistence_protect : 2; //!< sample count before the interrupt triggers
+    uint16_t interrupt_enable : 1;    //!< enable threshold interrupt
+    uint16_t shutdown : 1;            //!< set to 1 to shutdown the device, set to 0 to wakeup
+    uint16_t threshold_high;          //!< high threshold for the interrupt
+    uint16_t threshold_low;           //!< low threshold for the interrupt
+    uint16_t power_saving_mode : 2;   //!< power saving mode
+    uint16_t power_saving_enable : 1; //!< enable the pover saving mode
+} veml7700_config_t;
+
+/**
+ * @brief Initialize device descriptor
+ *
+ * Default SCL frequency is 100kHz. The I2C address is fix.
+ *
+ * @param dev Pointer to I2C device descriptor
+ * @param port I2C port number
+ * @param sda_gpio SDA GPIO
+ * @param scl_gpio SCL GPIO
+ * @return `ESP_OK` on success
+ */
+esp_err_t veml7700_init_desc(i2c_dev_t *dev, i2c_port_t port, gpio_num_t sda_gpio, gpio_num_t scl_gpio);
+
+/**
+ * @brief Free device descriptor
+ *
+ * @param dev Pointer to I2C device descriptor
+ * @return `ESP_OK` on success
+ */
+esp_err_t veml7700_free_desc(i2c_dev_t *dev);
+
+/**
+ * @brief Probe if the device exist on the bus
+ *
+ * @param dev Pointer to I2C device descriptor
+ * @return `ESP_OK` on success
+ */
+esp_err_t veml7700_probe(i2c_dev_t *dev);
+
+/**
+ * @brief Write the config to the device
+ *
+ * @param dev Pointer to I2C device descriptor
+ * @param config Pointer to the config descriptor
+ * @return `ESP_OK` on success
+ */
+esp_err_t veml7700_config(i2c_dev_t *dev, veml7700_config_t *config);
+
+/**
+ * @brief Read ambient light sensor value from the device
+ *
+ * @param dev Pointer to I2C device descriptor
+ * @param config Pointer to the config descriptor
+ * @param value_lux Pointer as return value in lux
+ * @return `ESP_OK` on success
+ */
+esp_err_t veml7700_ambient_light(i2c_dev_t *dev, veml7700_config_t *config, uint32_t *value_lux);
+
+/**
+ * @brief Read white channel value from the device
+ *
+ * @param dev Pointer to I2C device descriptor
+ * @param config Pointer to the config descriptor
+ * @param value_lux Pointer as return value in lux
+ * @return `ESP_OK` on success
+ */
+esp_err_t veml7700_white_channel(i2c_dev_t *dev, veml7700_config_t *config, uint32_t *value_lux);
+
+/**
+ * @brief Read the interrupt status from the device
+ *
+ * @param dev Pointer to I2C device descriptor
+ * @param low_threshold Pointer to return the low threshold passed indicator
+ * @param high_threshold Pointer to return the high threshold passed indicator
+ * @return `ESP_OK` on success
+ */
+esp_err_t veml7700_interrupt_status(i2c_dev_t *dev, bool *low_threshold, bool *high_threshold);
```

Comment:
Same: no verb in name, no reverse function.

Reply:
added a _get, can only be read so no reverse function.